nncf/torch/hardware/fused_patterns.py

```diff
@@ -13,49 +30,124 @@
+    return create_fc_conv_mul()
+
+
+def linear_operations():
+    pattern = GraphPattern()
+    pattern.add_node(**LINEAR_OPERATIONS)
+    return pattern
+
+def arithmetic_operations():
+    pattern = GraphPattern()
+    pattern.add_node(**ARITHMETIC_OPERATIONS)
+    return pattern
+
+
+def batch_norm_operations():
+    pattern = GraphPattern()
+    pattern.add_node(**BATCH_NORMALIZATION_OPERATIONS)
+    return pattern
+
+
+def activation_operations():
     atomic_activations = GraphPattern()
     atomic_activations.add_node(**ATOMIC_ACTIVATIONS_OPERATIONS)
     swish = create_swish_act()
     h_sigmoid = create_h_sigmoid_act()
     h_swish = create_h_swish_act()
-    activations = atomic_activations | swish | h_swish | h_sigmoid
-    retval.register(activations, 'ACTIVATIONS', match=False)
 
-    arithmetic_ops = GraphPattern()
-    arithmetic_ops.add_node(**ARITHMETIC_OPERATIONS)
-    retval.register(arithmetic_ops, ARITHMETIC_OPERATIONS['label'], match=False)
+    pattern = GraphPattern()
+    pattern.add_pattern_alternative(atomic_activations)
+    pattern.add_pattern_alternative(swish)
+    pattern.add_pattern_alternative(h_swish)
+    pattern.add_pattern_alternative(h_sigmoid)
+    return pattern
 
-    batch_norm_activations_permutation = batch_norm + activations | activations + batch_norm | batch_norm | activations
 
-    retval.register(linear_ops + arithmetic_ops, 'LINEAR + ARITHMETIC', match=True)
-    retval.register(linear_ops + batch_norm_activations_permutation, 'LINEAR + BN_ACT_PERM',
-                    match=True)
-    retval.register(batch_norm + activations, 'BN + ACTIVATIONS', match=True)
-    retval.register(activations + batch_norm, 'ACTIVATIONS + BN', match=True)
-    retval.register(arithmetic_ops + batch_norm_activations_permutation,
-                    'ARITHMETIC + BN_ACT_PERM', match=True)
+def batch_norm_activations_permutations():
+    batch_norm = batch_norm_operations()
+    activations = activation_operations()
 
-    group_norm = GraphPattern()
-    group_norm.add_node(**GROUP_NORMALIZATION_OPERATIONS)
-    relu = GraphPattern()
-    relu.add_node(**RELU_OPERATIONS)
-    retval.register(group_norm + relu, 'GROUP_NORM + RELU', match=True)
+    bn_act = GraphPattern()
+    bn_act.add_pattern_alternative(batch_norm)
+    bn_act.join_patterns(activations)
+
+    act_bn = GraphPattern()
+    act_bn.add_pattern_alternative(activations)
+    act_bn.join_patterns(batch_norm)
 
-    l2_norm = create_l2_norm()
-    retval.register(l2_norm, 'L2_NORM', match=True)
-    fc_mul = create_fc_conv_mul()
-    retval.register(fc_mul, 'FC_MUL_CONST', match=True)
-    return retval
+    pattern = GraphPattern()
+    pattern.add_pattern_alternative(bn_act)
+    pattern.add_pattern_alternative(act_bn)
+    pattern.add_pattern_alternative(batch_norm)
+    pattern.add_pattern_alternative(activations)
+    return pattern
 
 
-PT_HW_FUSED_PATTERNS = _get_torch_hw_fused_patterns()
+def get_torch_hw_patterns():
```

Comment:
Why do we need this method? Why not use `PatternsManager` as is?

Reply:
I did it to simplify code inside torch algorithms: instead of importing BackendType, TargetDevice  and using them every time I made a default version. I believe this approach is easier to understand as Torch has no target specific patterns and logic yet